datasetUtils/getDatasetDetailsPublished.go

```diff
@@ -30,58 +39,72 @@ func GetDatasetDetailsPublished(client *http.Client, APIServer string, datasetLi
 		if end > len(datasetList) {
 			end = len(datasetList)
 		}
-
-		var filter = `{"where":{"pid":{"inq":["` +
-			strings.Join(datasetList[i:end], `","`) +
-			`"]}},"fields":{"pid":true,"sourceFolder":true,"size":true,"ownerGroup":true,"numberOfFiles":true}}`
-
-		v := url.Values{}
-		v.Set("filter", filter)
-		v.Add("isPublished", "true")
-
-		var myurl = APIServer + "/Datasets?" + v.Encode()
-		// log.Println("Url:", myurl)
-
-		resp, err := client.Get(myurl)
+		
+		filter := createFilter(datasetList[i:end])
+		
+		resp, err := makeRequest(client, APIServer, filter)
 		if err != nil {
 			log.Fatal("Get dataset details failed:", err)
 		}
 		defer resp.Body.Close()
-
+		
 		if resp.StatusCode == 200 {
-			body, _ := ioutil.ReadAll(resp.Body)
+			outputDatasetDetails, urls := processDatasetDetails(resp, datasetList[i:end])
+			return outputDatasetDetails, urls
+		} else {
+			log.Printf("Querying dataset details failed with status code %v\n", resp.StatusCode)
+		}
+	}
+			
+	return nil, nil
+}
 
-			datasetDetails := make([]Dataset, 0)
+func createFilter(datasetList []string) string {
+	return `{"where":{"pid":{"inq":["` +
+	strings.Join(datasetList, `","`) +
+	`"]}},"fields":{"pid":true,"sourceFolder":true,"size":true,"ownerGroup":true,"numberOfFiles":true}}`
+}
 
-			_ = json.Unmarshal(body, &datasetDetails)
+func makeRequest(client *http.Client, APIServer string, filter string) (*http.Response, error) {
+	v := url.Values{}
+	v.Set("filter", filter)
+	v.Add("isPublished", "true")
+	
+	var myurl = APIServer + "/Datasets?" + v.Encode()
+	
+	return client.Get(myurl)
+}
 
-			// verify if details were actually found for all available Datasets
-			for _, datasetId := range datasetList[i:end] {
-				detailsFound := false
-				for _, datasetDetail := range datasetDetails {
-					if datasetDetail.Pid == datasetId {
-						detailsFound = true
-						outputDatasetDetails = append(outputDatasetDetails, datasetDetail)
-						color.Set(color.FgGreen)
-						log.Printf("%s %9d %v %v\n", datasetId, datasetDetail.Size/1024./1024., datasetDetail.OwnerGroup, datasetDetail.SourceFolder)
-						color.Unset()
-						//https: //doi2.psi.ch/datasets/das/work/p16/p16628/20181012_lungs/large_volume_360/R2-6/stitching/data_final_volume_fullresolution/
-						url := "https://" + PUBLISHServer + "/datasets" + datasetDetail.SourceFolder
-						urls = append(urls, url)
-						sizeArray = append(sizeArray, datasetDetail.Size)
-						numFilesArray = append(numFilesArray, datasetDetail.NumberOfFiles)
-						break
-					}
-				}
-				if !detailsFound {
-					color.Set(color.FgRed)
-					log.Printf("Dataset %s no infos found in catalog - will not be copied !\n", datasetId)
-					color.Unset()
-				}
+func processDatasetDetails(resp *http.Response, datasetList []string) ([]Dataset, []string) {
```

Comment:
isn't this block very similar to [this](https://github.com/paulscherrerinstitute/scicat-cli/pull/65/files#diff-1aba7554ddcdff9d1dfdba412e079d59438b7ef31ae25f09f3239b7cdd854e9eR65-R86)? If so, in a later PR, we can reuse a common function